### DISCUSSION TITLE
I don't understand github actions at all

### DISCUSSION THREAD

**Discussion by EwonRael** [indestructible-type/Besley#21] [General]
@aaronbell I couple months ago I pushed the changes you made so it uses github actions. Just now I undid those changes, because I want to do some work on maintaining the font and I have zero clue how to use github actions.

In general I'm not a coder, I'm more of a font guy with some very (very) basic understanding of code. I feel guilty because I shouldn't of accepted the request in the first place.

I'm not against using this technology. I just don't know how. I've done some reading on it and I feel way over my head. If there's some great advantage to it, or you feel it's necessary, I'm happy to chatting and re-implementing it if/when I understand how to use it.

Hope you are well in all ways.

That is all

**Comment by aaronbell** [comment]
Hi Owen,

No worries at all. In all honesty I don’t totally understand the code behind GitHub actions either, but the GF team has done a good job at making it pretty easy to at least understand the input / output. Let me see if I can help explain it.

Essentially there are two parts to the new build process. 

First is the makefile that lives in the main folder of the repro. This is the file that runs the scripts that generate the font. It will pull in your UFOs, process them, and spit out TTFs into the fonts folder. You can actually run this file locally by entering “make build” while in the folder from the command prompt. It will create the virtual environment and do everything (you may find you need to delete the build.stamp file though if you’ve run it before). That’s pretty easy!

Second is the GitHub action itself. The GitHub action, on PR, will run the makefile script and record the process in the “Actions” tab (https://github.com/indestructible-type/Besley/actions). And the fonts created, instead of going in a fonts folder (IIRC), are exported as “artifacts” of a given action (see the bottom of here: https://github.com/indestructible-type/Besley/actions/runs/1020406577). You can then upload them as new versions of the font. So all of the processing is handled server side.

Ultimately the idea is that you, as the designer, really only need to worry about your font sources. “Make build” will take care of everything else. 

If you don’t feel comfortable with all of this, that’s just fine. It is your font and you should do what you’d like with it! Do keep in mind that if you’d like to update the version on GF you’ll need the STAT table fix that I had implemented.

Let me know if I can be of more assistance!
